### PR TITLE
contrib: Add test for ELF symbol-check

### PR DESCRIPTION
**Pull request by laanwj**
Check both failure cases:
- Use a glibc symbol from a version that is too new
- Use a symbol from a library that is not in the allowlist

And also check a conforming binary.

Adding a similar check for Windows PE can be done in a separate PR.

### CODE REVIEW

File: contrib/devtools/test-symbol-check.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        source = 'test1.c'
+        executable = 'test1'
+        cc = 'gcc'
+
+        # there's no way to do this test for RISC-V at the moment; bionic's libc is 2.27
+        # and we allow all symbols from 2.27.
+        if 'riscv' in get_machine(cc):
+            self.skipTest("test not available for RISC-V")
+
+        # emfd_create was introduced in GLIBC 2.27, so is newer than the upper limit of
```
Comment (fanquake) [inline contrib/devtools/test-symbol-check.py:31]: s/emfd_create/memfd_create

**Review by fanquake — COMMENTED** [review]
Concept ACK - thanks for putting this together. Will actually test shortly.

**Comment by fanquake** [comment]
I added a test for macOS if you wanted to pull that in as well: https://github.com/fanquake/bitcoin/commit/9e6e59b01e1e4e3325bab59cf97139b462b42651.

**Comment by laanwj** [comment]
> I added a test for macOS if you wanted to pull that in as well: fanquake@9e6e59b.

Thanks ! I didn't mention macOS in the OP because I (mistakenly) thought we only had a symbol check for windows and linux. Good to have a test for that too.

I've fixed the comment and cherry-picked that commit.

**Comment by laanwj** [comment]
(close/ropen was to test the IRC bot)

**Review by fanquake — APPROVED** [review]
ACK ed1bbcefeaafef460b6e5609de85e13c12e0d2a4

**Pull request merged**